pkg/apis/core/v1/helper/helpers.go

```diff
@@ -36,3 +36,22 @@ func AddToNodeAddresses(addresses *[]v1.NodeAddress, addAddresses ...v1.NodeAddr
 		}
 	}
 }
+
+// RemoveFromNodeAddresses removes the NodeAddresses from the passed-by-pointer
+// slice if they already exist.
+func RemoveFromNodeAddressees(addresses *[]v1.NodeAddress, removeAddresses ...v1.NodeAddress) {
```

Comment:
Can we move this function to `pkg/cloudprovider/providers/openstack/openstack_instances.go`? `AddToNodeAddresses` can live here as because it's not related to this PR.

Reply:
Not sure why do you want to leave one helper, but move out a new one. They both are definitely helpers, so why split?

Reply:
@kayrus If you could, please move out all the functions from the `helpers.go`. We don't need that file for utility functions. For the two specific functions, we could put in either `openstack.go`.

Reply:
thanks, this makes more sense.

Reply:
@lingxiankong in addition to your proposal, does it make sense to move `pkg/api/v1/service/util.go` functions into `pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go`?

Reply:
It's not necessary, up to you.

Reply:
I'll move it then